customer-service/omnichannel/real-time-translation/webResourceV2.js

```diff
@@ -9,464 +9,695 @@ If you use the sample script and enable data collection in your applications, yo
+          ]; //replacing the current code with original C1 language code which came during initialization of new conversation
+        value.destinationLanguage =
+          C1WebResourceNamespace.dictForAllConversation[conversationId][
+            "C1LangLocaleCode"
+          ]; //replacing the current code with original C1 language code which came during initialization of new conversation
+      } else if (
+        translateToC1orC2 ==
+        Microsoft.Omnichannel.TranslationFramework.TranslateTo.C1
+      ) {
+        value.destinationLanguage =
+          C1WebResourceNamespace.dictForAllConversation[conversationId][
+            "C1LangLocaleCode"
+          ]; //replacing the current code with original C1 language code which came during initialization of new conversation
+      } else if (
+        translateToC1orC2 ==
+        Microsoft.Omnichannel.TranslationFramework.TranslateTo.C2
+      ) {
+        value.sourceLanguage =
+          C1WebResourceNamespace.dictForAllConversation[conversationId][
+            "C1LangLocaleCode"
+          ]; //replacing the current code with original C1 language code which came during initialization of new conversation
+      }
+      return value;
+    });
+    return response;
+  },
+
+  //https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-translate
+  translateMessageInternalAzure: async function (
+    conversationId,
+    message,
+    messageSender,
+    sourceLang,
+    destLang
+  ) {
+    var errorObj = {
+      isError: false,
+      errorCode: null,
+    };
+    //when we do not have even a single clue about C2's lang and c1 wants to send him message so destlang == null
+    if (destLang == null || sourceLang == destLang)
+      return {
+        translatedMessage: message,
+        destinationLanguage:
+          C1WebResourceNamespace.getOcLanguageCodeMapFromISO6391LanguageCode(
+            String(sourceLang)
+          ),
+        errorObject: errorObj,
+        sourceLanguage:
+          C1WebResourceNamespace.getOcLanguageCodeMapFromISO6391LanguageCode(
+            String(sourceLang)
+          ),
+      };
+
+    var url = "";
+    if (sourceLang == null) {
+      url =
+        "https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&to=" +
+        destLang;
+    } else {
+      url =
+        "https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&from=" +
+        sourceLang +
+        "&to=" +
+        destLang;
+    }
+    //translation happens here by calling api for translation engine
+    var myJson = null;
+    try {
+      var bodyObj = [];
+      bodyObj[0] = new Object();
+      bodyObj[0].Text = message;
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify(bodyObj), // string or object
+        headers: {
+          "Content-Type": "application/json; charset=UTF-8",
+          "Authorization":
+            `Bearer ${C1WebResourceNamespace.authToken}`,
```

Comment:
Don't we support the secret approach at all? we should still keep the sample around, right?

Reply:
Have added this as a separate file for implementation with translator v3 so we can keep the original sample around.